packages/daimo-userop/src/daimoOpSender.ts

```diff
@@ -12,56 +12,94 @@ import {
 
 import { DaimoOpBuilder, DaimoOpMetadata } from "./daimoOpBuilder";
 import { SigningCallback } from "./signingCallback";
-import config from "../config.json";
+
+interface DaimoOpConfig {
+  /// Chain ID
+  chainId: number;
+  /// Bundler RPC URL.
+  bundlerRpcUrl: string;
+  /// Stablecoin token address.
+  tokenAddress: Address;
+  /// Decimals for that token.
+  tokenDecimals: number;
+  /// EphemeralNotes instance. The stablecoin used must match tokenAddress.
+  notesAddress: `0x${string}`;
+  /// Paymaster, payable in tokenAddress.
+  paymasterAddress: Address;
+  /// Daimo account address.
+  accountAddress: Address;
+  /// Signs userops. Must, in some form, check user presence.
+  accountSigner: SigningCallback;
+}
 
 /// DaimoOpSender constructs user operations for a Daimo account.
 /// Supports key rotations, token transfers, and ephemeral note ops.
 export class DaimoOpSender {
   /** Connection to the chain */
   private provider: BundlerJsonRpcProvider;
 
-  private opBuilder: DaimoOpBuilder;
-
-  private tokenAddress: Address;
-  private tokenDecimals: number;
-
-  private notesAddress: `0x${string}`;
-
-  constructor(
-    /// Userop builder, containing bundler and paymaster configuration.
-    _opBuilder: DaimoOpBuilder,
-    /// Stablecoin token address.
-    _tokenAddress: `0x${string}`,
-    /// Decimals for that token.
-    _tokenDecimals: number,
-    /// EphemeralNotes instance. The stablecoin used must match _tokenAddress.
-    _notesAddress: `0x${string}`
+  private constructor(
+    private opConfig: DaimoOpConfig,
+    private opBuilder: DaimoOpBuilder
   ) {
-    this.provider = new BundlerJsonRpcProvider(config.bundlerRpcUrl);
-    this.opBuilder = _opBuilder;
+    this.provider = new BundlerJsonRpcProvider(opConfig.bundlerRpcUrl);
+  }
 
-    this.tokenAddress = _tokenAddress;
-    this.tokenDecimals = _tokenDecimals;
+  /**
+   * Initializes using the DAIMO_CHAIN and DAIMO_BUNDLER_RPC env vars.
+   */
+  public static async initFromEnv(
+    accountAddress: Address,
+    accountSigner: SigningCallback
+  ): Promise<DaimoOpSender> {
+    const { tokenAddress, tokenDecimals, paymasterAddress, chainL2 } =
+      Contracts.chainConfig;
 
-    this.notesAddress = _notesAddress;
+    let bundlerRpcUrl = process.env.DAIMO_BUNDLER_RPC || "";
+    if (bundlerRpcUrl === "" && chainL2.network === "base-goerli") {
+      bundlerRpcUrl = // Default to testnet
+        "https://api.pimlico.io/v1/base-goerli/rpc?apikey=70ecef54-a28e-4e96-b2d3-3ad67fbc1b07";
```

Comment:
looks like the API key Pimlico/Kristoff mentioned we should rotate? do you want to remove this before we go to mainnet?

Reply:
It's been on GH for a few weeks

We have do a bit more than just rotate--need to proxy thru our API. Will do in a followup pr